src/oscar/apps/customer/alerts/utils.py

```diff
@@ -1,134 +1,126 @@
 import logging
 
-from django.contrib.sites.models import Site
-from django.core import mail
 from django.db.models import Max
 from django.template import loader
 
-from oscar.apps.customer.notifications import services
 from oscar.core.loading import get_class, get_model
 
-CommunicationEventType = get_model('customer', 'CommunicationEventType')
 ProductAlert = get_model('customer', 'ProductAlert')
 Product = get_model('catalogue', 'Product')
-Dispatcher = get_class('customer.utils', 'Dispatcher')
+Dispatcher = get_class('communication.utils', 'Dispatcher')
 Selector = get_class('partner.strategy', 'Selector')
 
-logger = logging.getLogger('oscar.alerts')
+alerts_logger = logging.getLogger('oscar.alerts')
 
 
-def send_alerts():
+class AlertsDispatcher:
     """
-    Send out product alerts
+    Dispatcher to send concrete product alerts related emails
+    and notifications.
     """
-    products = Product.objects.filter(
-        productalert__status=ProductAlert.ACTIVE
-    ).distinct()
-    logger.info("Found %d products with active alerts", products.count())
-    for product in products:
-        send_product_alerts(product)
 
-
-def send_alert_confirmation(alert):
-    """
-    Send an alert confirmation email.
-    """
-    ctx = {
-        'alert': alert,
-        'site': Site.objects.get_current(),
-    }
-
-    code = 'PRODUCT_ALERT_CONFIRMATION'
-    messages = CommunicationEventType.objects.get_and_render(code, ctx)
-
-    if messages and messages['body']:
-        Dispatcher().dispatch_direct_messages(alert.email, messages)
-
-
-def send_product_alerts(product):   # noqa C901 too complex
```

Comment:
Can we make this backwards compatible by leaving it here for now, adding a deprecation warning and then calling `AlertsDispatcher().send_product_alert_email_for_user(instance.product)`?

Reply:
Added for all methods in `src/oscar/apps/customer/alerts/utils.py`:
* `send_alerts`
* `send_alert_confirmation`
* `send_product_alerts`